### PR TITLE
Parlem SMS integration

### PR DESCRIPTION
**Pull request by laurajaime**

### CODE REVIEW

**Review by NicoleSo13 — APPROVED** [review]
LGTM

File: decidim-verifications-sms_direct/CHANGELOG.MD [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# CHANGELOG
+
+## 0.0.1
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/CHANGELOG.MD:3]: any comments here?

File: decidim-verifications-sms_direct/app/models/decidim/verifications/sms_direct/phone_code.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        EXPIRATION_DAYS= 1
+
+        belongs_to :organization, foreign_key: :decidim_organization_id,
+                                class_name: "Decidim::Organization"
+
+        before_save :normalize_phone
+
+        scope :expired, -> { where("updated_at < ?", Time.zone.now - EXPIRATION_DAYS.days) }
+
+        # An organzation scope
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/app/models/decidim/verifications/sms_direct/phone_code.rb:16]:
```suggestion
        # An organization scope
```

File: decidim-verifications-sms_direct/app/packs/src/decidim/verifications/sms_direct/application.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  verificationCodeInput.disabled = true;
+
+  const sendSmsCodeButton = document.querySelector('.send-sms-code');
+  
+  sendSmsCodeButton.addEventListener('click', function() {
+    sendSmsCodeButton.style.display = 'none';
+    mobilePhoneInput.readOnly = true;
+
+    let mobilePhoneNumber = mobilePhoneInput.value;
+    alert("Validate phone number with regexp")
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/app/packs/src/decidim/verifications/sms_direct/application.js:13]: Be careful!!

File: decidim-verifications-sms_direct/app/packs/src/decidim/verifications/sms_direct/application.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    let request = new XMLHttpRequest();
+    var url = `/verifications_sms_direct/sms_codes?phone_num=${mobilePhoneNumber}`;
+    request.open("POST", url, true);
+ 
+    request.onreadystatechange = function () {
+      if (this.readyState == 4 && this.status == 200) {
+        console.log(this.response)
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/app/packs/src/decidim/verifications/sms_direct/application.js:21]: do you want to keep this?

File: decidim-verifications-sms_direct/decidim-verifications-sms_direct.gemspec [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+require "decidim/verifications/sms_direct/version"
+
+# Describe your gem and declare its dependencies:
+Gem::Specification.new do |s|
+  s.name = "decidim-verifications-sms_direct"
+  s.version = Decidim::Verifications::SmsDirect::VERSION
+  s.authors = ["Laura Jaime", "Oliver Valls"]
+  s.email = ["laura.jv@coditramuntana.com"]
+  s.summary = "SMS code verifier"
+  # s.homepage = "calafell"
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/decidim-verifications-sms_direct.gemspec:15]: can we change it?

File: decidim-verifications-sms_direct/spec/controllers/decidim/sms_verifier/sms_verifier_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+RSpec.describe Decidim::Verifications::SmsDirect::SmsCodesController, type: :controller do
+  # include Warden::Test::Helpers
```
Comment (laurajaime) [inline decidim-verifications-sms_direct/spec/controllers/decidim/sms_verifier/sms_verifier_controller_spec.rb:5]: can we remove it?

**Review by laurajaime — COMMENTED** [review]
Awesome job @tramuntanal !

Only some little things..

**Comment by tramuntanal** [comment]
All changes applied and tests succeeded @laurajaime

**Pull request merged**